### PR TITLE
Do not show edited timestamp details if it duplicates created timestamp

### PR DESCRIPTION
**Pull request by lyzadanger**
Only show the relative time-since string for the annotation edited date
if it is different from the relative time-since string for the created
date. If they are the same, just show "edited" without further detail.

Before:

image

After:

image

Before, we had a somewhat abstract `Timestamp` component. It was nice to be able to treat the created and edited dates as two `Timestamp` components, but this requirement makes them interdependent. This PR adds a new `AnnotationTimestamps` component and does away with the `Timestamp` component.

Fixes https://github.com/hypothesis/product-backlog/issues/1156

### CODE REVIEW

**Review by robertknight — CHANGES_REQUESTED** [review]
I noticed a couple of incomplete tests and suggested some small optimizations. Otherwise the change seems good.

File: src/sidebar/components/annotation-timestamps.js [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+ * for this annotation).
+ *
+ * @param {AnnotationTimestampsProps} props
+ */
+export default function AnnotationTimestamps({
+  annotation,
+  withEditedTimestamp,
+}) {
+  // "Current" time, used when calculating the relative age of `timestamp`.
+  const [now, setNow] = useState(new Date());
```
[the remark this answers is not in the document]
Comment (lyzadanger) [reply src/sidebar/components/annotation-timestamps.js]: Ah, in defense this was a C'nP from the previous implementation of `Timestamp`. Fixed in any case.

File: src/sidebar/components/annotation-timestamps.js [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+  const createdDate = useMemo(() => new Date(annotation.created), [annotation]);
```
[the remark this answers is not in the document]
Comment (lyzadanger) [reply src/sidebar/components/annotation-timestamps.js]: Done

File: src/sidebar/components/annotation-timestamps.js [hunk 3]
```diff
@@ -0,0 +1,115 @@
...
+      relative: toFuzzyString(updatedDate, now),
+    };
+  }, [updatedDate, now]);
+
+  // Refresh relative timestamp, at a frequency appropriate for the age.
+  useEffect(() => {
+    // Determine which of the two Dates to use for the `decayingInterval`
+    // It should be the latest relevant date, as the interval will be
+    // shorter the closer the date is to "now"
+    const laterDate =
```
[the remark this answers is not in the document]
Comment (lyzadanger) [reply src/sidebar/components/annotation-timestamps.js]: Indeed. Done.

File: src/sidebar/components/test/annotation-header-test.js [hunk 4]
```diff
@@ -164,39 +164,35 @@ describe('AnnotationHeader', () => {
   });
 
   describe('timestamps', () => {
-    it('should render timestamp container element if annotation has a `created` value', () => {
+    it('should not render timestamps if annotation is under edit');
```
[the remark this answers is not in the document]
Comment (lyzadanger) [reply src/sidebar/components/test/annotation-header-test.js]: Derp deee derp derp. Thanks for the catch.

File: src/sidebar/components/test/annotation-timestamps-test.js [hunk 5]
```diff
@@ -0,0 +1,116 @@
...
+
+  it('renders a linked created timestamp if annotation has a link', () => {
+    const annotation = fixtures.defaultAnnotation();
+    annotation.links = { html: 'http://www.example.com' };
+
+    const wrapper = createComponent({ annotation });
+
+    const link = wrapper.find('a.annotation-timestamps__created');
+    const linkEl = link.getDOMNode();
+    assert.equal(linkEl.getAttribute('href'), 'http://www.example.com');
```
[the remark this answers is not in the document]
Comment (lyzadanger) [reply src/sidebar/components/test/annotation-timestamps-test.js]: Ah! TIL. Changed.

**Pull request merged**